rclc/include/rclc/executor.h

```diff
@@ -0,0 +1,387 @@
+ * Allocates Memory   | No
+ * Thread-Safe        | No
+ * Uses Atomics       | No
+ * Lock-Free          | Yes
+ *
+ * \param [inout] executor pointer to initialized executor
+ * \param [in] timer pointer to an allocated timer
+ * \return `RCL_RET_OK` if add-operation was successful
+ * \return `RCL_RET_INVALID_ARGUMENT` if any parameter is a null pointer
+ * \return `RCL_RET_ERROR` if any other error occured
+ */
+rcl_ret_t
+rclc_executor_add_timer(
+  rclc_executor_t * executor,
+  rcl_timer_t * timer);
+
+/**
+ *  The spin-some function checks one-time for new data from the DDS-queue.
+ * * the timeout is defined in {@link rclc_executor_t.timeout_ns} and can
+ *   be set by calling {@link rclc_executor_set_timeout()} function (default value is 100ms)
+ *
+ * The static-LET executor performs the following actions:
+ * * initializes the wait_set with all handle of the array executor->handles
+ * * waits for new data from DDS queue with rcl_wait() with timeout executor->timeout_ns
+ * * takes all ready handles from the wait_set with rcl_take()
+ * * processes all handles in the order, how they were added to the executor with the respective add-functions
+ *   by calling respective callback (thus implementing first-read, process, semantic of LET)
+ *
+ * Memory is dynamically allocated within rcl-layer, when DDS queue is accessed with rcl_wait_set_init()
+ *
+ * <hr>
+ * Attribute          | Adherence
+ * ------------------ | -------------
+ * Allocates Memory   | Yes
+ * Thread-Safe        | No
+ * Uses Atomics       | No
+ * Lock-Free          | Yes
+ *
+ *
+ * \param [inout] executor pointer to initialized executor
+ * \param[in] timeout_ns  timeout in millisonds
+ * \return `RCL_RET_OK` if spin_once operation was successful
+ * \return `RCL_RET_INVALID_ARGUMENT` if any parameter is a null pointer
+ * \return `RCL_RET_TIMEOUT` if rcl_wait() returned timeout (aka no data is avaiable during until the timeout)
+ * \return `RCL_RET_ERROR` if any other error occured
+ */
+rcl_ret_t
+rclc_executor_spin_some(
+  rclc_executor_t * executor,
+  const uint64_t timeout_ns);
+
+/**
+ *  The spin function checks for new data at DDS queue as long as ros context is available.
+ *  It calls {@link rclc_executor_spin_some()} as long as rcl_is_context_is_valid() returns true.
+ *
+ *  Memory is dynamically allocated within rcl-layer, when DDS queue is accessed with rcl_wait_set_init()
+ *  (in spin_some function)
+ *
+ * <hr>
+ * Attribute          | Adherence
+ * ------------------ | -------------
+ * Allocates Memory   | Yes
+ * Thread-Safe        | No
+ * Uses Atomics       | No
+ * Lock-Free          | Yes
+ *
+ *
+ * \param [inout] executor pointer to initialized executor
+ * \return `RCL_RET_OK` if spin operation was successful
+ * \return `RCL_RET_INVALID_ARGUMENT` if executor is a null pointer
+ * \return `RCL_RET_ERROR` if any other error occured
+ */
+rcl_ret_t
+rclc_executor_spin(rclc_executor_t * executor);
+
+
```

Comment:
```suggestion
```

Reply:
fixed in 54f0bbc54a33cc601764ab1d53fe3119d9e61c8b